### ISSUE TITLE
Unable to load DLL 'libwkhtmltox' or one of its dependencies

### ISSUE THREAD

**Issue by Schecher1** [imaxs/HTMLconvert#1]
Hey,
I was looking for a HTML => PNG converter for C# and was very happy when I saw this. Everything looks good except when I get to the "imageConverter.Convert(new ImageSettings()" it throws an exception "System.DllNotFoundException". 

![image](https://user-images.githubusercontent.com/74829802/157334085-8feb7bdc-564f-47e9-a521-57c590375554.png)

**Comment by imaxs** [comment]
Hello,
This is a fairly common issue.

Windows reference - libwkhtmltox.dll
OSX reference - libwkhtmltox.dylib
Linux reference - libwkhtmltox.os

Add it to your project reference by custom Assembly reference & register in your startup project.

**Comment by Schecher1** [comment]
Hello,

yes actually i thought about that and tested it out, but i get a VS Error (that's german and in english it says) "The reference is invalid or not supported".

I use VS2022, NET6 and as console.

I have tested the 32bit and the 64bit version 

The files were in "HTMLconvert/HTMLconvert/Lib/".


any suggestions?


greetings
![image](https://user-images.githubusercontent.com/74829802/159536846-7c507d21-36fc-4acc-af79-c7be4891a4e9.png)

**Comment by imaxs** [comment]
> The files were in "HTMLconvert/HTMLconvert/Lib/".
> any suggestions?

Please make sure that the wkhtmltopdf dependencies are enabled correctly.

[https://stackoverflow.com/questions/13791196/using-wkhtmltopdf-on-windows](https://stackoverflow.com/questions/13791196/using-wkhtmltopdf-on-windows)

**Comment by Schecher1** [comment]
ur link is broken :D 
![image](https://user-images.githubusercontent.com/74829802/160296539-6410ef61-ebe0-41a7-bd29-15b0dbed59a3.png)

**Comment by imaxs** [comment]
> ur link is broken :D ![image](https://user-images.githubusercontent.com/74829802/160296539-6410ef61-ebe0-41a7-bd29-15b0dbed59a3.png)

Hmm.. Must be some kind of github bug.

Here is the link: https://stackoverflow.com/questions/13791196/using-wkhtmltopdf-on-windows

Have you tried setting libwkhtmltox library dependencies manually?

https://wkhtmltopdf.org/downloads.html

**Comment by Schecher1** [comment]
No, I just downloaded the file and added it under "Add project reference". Or what ur mean 😅

**Comment by imaxs** [comment]
> No, I just downloaded the file and added it under "Add project reference". Or what ur mean 😅

Ah, so you need to install wkhtmltopdf first

**Comment by Schecher1** [comment]
oh where?

**Comment by Schecher1** [comment]
oh i have thought all the time that i only have to import the dll... oops

**Comment by Schecher1** [comment]
I can only find links for linux, but I would also like to debug it on windows etc.

**Comment by imaxs** [comment]
> oh i have thought all the time that i only have to import the dll... oops

you are right
But the wrapper for wkhtmltopdf doesn't see the dll. In this case it should be specified directly by changing the code.

> I can only find links for linux, but I would also like to debug it on windows etc.

Unfortunately I didn't check the correct execution for Windows.
Perhaps one of these days I will release an update that will make the wrapper work correctly on Windows.

**Comment by Schecher1** [comment]
that means it would work under linux? hm too bad, i don't really feel like developing on linux now. But Thanks, i´ll wait for the Windows Version
